### ISSUE TITLE
Feedreader is unable to login and retrieve articles

### ISSUE THREAD

**Issue by okias** [jangernert/FeedReader#185]
Feedreader started behave weird after purging DB. It stops on login screen, that password is bad (it's correct), after while switch to "OFFLINE" mode with rotating wheel... no articles loaded anymore.
```
$ feedreader
[ INFO  ] Everything is logged. Used to debug!
[ INFO  ] FeedReader 1.6 beta
[ DEBUG ] MainWindow: setupCSS
[WARNING] dark.css:57:18Missing name of pseudo-class
[ DEBUG ] ContentPage: setup FeedList
[ DEBUG ] UpdateButton: update status
[ DEBUG ] UpdateButton: setSensitive true
[ DEBUG ] MainWindow: determining state
[ DEBUG ] SELECT count(*) FROM main.articles
[ DEBUG ] count 0
[ DEBUG ] SELECT count(*) FROM main.categories
[ DEBUG ] count 0
[ DEBUG ] SELECT count(*) FROM main.feeds
[ DEBUG ] count 0
[ DEBUG ] SELECT count(*) FROM main.tags
[ DEBUG ] count 0
[ DEBUG ] MainWindow: show login

** (feedreader:7548): CRITICAL **: secret_schema_unref: assertion 'g_atomic_int_get (&schema->reserved) > 0' failed

** (feedreader:7548): CRITICAL **: secret_schema_unref: assertion 'g_atomic_int_get (&schema->reserved) > 0' failed
[ DEBUG ] MainWindow: show error bar - errorCode = 6
[ DEBUG ] MainWindow: dont show error bar
[ DEBUG ] HeaderBar: updatebutton status false
[ DEBUG ] UpdateButton: setSensitive false
[ DEBUG ] DBusConnection: setOffline
[ DEBUG ] FeedList: new FeedList
[ DEBUG ] initCollapseCategories
[ DEBUG ] FeedList: restore selected row: feed -4
[ DEBUG ] ArticleList: newList
[ DEBUG ] ArticleList: create
[ DEBUG ] ArticleList: offset 0
[ DEBUG ] limit: 20
[ DEBUG ] load articles from db
[ DEBUG ] SELECT ROWID, feedID, articleID, title, author, url, preview, unread, marked, tags, date, guidHash FROM main.articles WHERE unread = 9  ORDER BY rowid DESC LIMIT 20 OFFSET 0
[ DEBUG ] actual articles loaded: 0
[ DEBUG ] ArticleList: insert new rows
[ DEBUG ] ArticleList: create finished
[ DEBUG ] write login data
[ DEBUG ] LoginPage: status = FEED_READER_LOGIN_RESPONSE_MISSING_PASSWD
[ DEBUG ] MainWindow: show error bar - errorCode = 2
[ DEBUG ] MainWindow: show error bar
[ DEBUG ] DBusConnection: setOffline
[ DEBUG ] FeedList: new FeedList
[ DEBUG ] initCollapseCategories
[ DEBUG ] FeedList: restore selected row: feed -4
[ DEBUG ] write login data
[ DEBUG ] LoginPage: status = FEED_READER_LOGIN_RESPONSE_MISSING_PASSWD
[ DEBUG ] MainWindow: show error bar - errorCode = 2
[ DEBUG ] MainWindow: show error bar
[ DEBUG ] DBusConnection: setOffline
[ DEBUG ] FeedList: new FeedList
[ DEBUG ] initCollapseCategories
[ DEBUG ] FeedList: restore selected row: feed -4

(feedreader:7548): Gtk-CRITICAL **: gtk_widget_event: assertion 'WIDGET_REALIZED_FOR_EVENT (widget, event)' failed
```

**Comment by jangernert** [comment]
The log of the UI doesn't help much in this case. I need to know what is happening in the daemon.
So please post the log of the feedreader-daemon instead.

**Comment by jangernert** [comment]
Btw: what exactly do you mean by "purging DB"?

**Comment by okias** [comment]
rm .local/share/feedreader/* -rf

Anyway, after running feedreader-daemon from cmd and running feedreader, it magically loads everything without error.... No, WORKSFORME, but not sure what error happend before.

**Comment by jangernert** [comment]
Did you restart the daemon after removing the local database?
I don't  know what happens inside the sqlite-library when the database is suddenly gone. But as long as the daemon is not restarted the connection to the (now non-existing) database should be still up.
This should theoretically not have anything to do with the error during the login.
If you want you can delete your data again and see if you can trigger the bug again.

**Comment by jangernert** [comment]
Btw: Was there any reason you removed the local user-data of FeedReader? Was something else not working for you?

**Comment by okias** [comment]
1. wow, I really never seen that caring from developers of OSS SW! Thank you!
2. yes, after switch from owncloud to nextcloud it somehow regenerated DB but then it started be really extremly slow and sometimes froze for half minute etc. so I wiped DB. 

3. I have feeling I didn't kill feedreader-daemon, but between trials have been few reboots, so maybe feedreader-daemon created some damaged DB..

**Comment by okias** [comment]
4. anyway now it still shows activity, but doesn't load new articles :( still in "waiting state" to update

only errors I see in logs are:
[ ERROR ] Grabber: config file has no rule for 'body'
[ DEBUG ] GrabberUtils: save Images: 1076, 19
[ ERROR ] Grabber: no body found

**Comment by okias** [comment]
maybe something UTF8 f*ckup in glib?

**Comment by jangernert** [comment]
oO are those the actual URL's of the articles?
Seems like the content-grabber can't handle it. You can deactivate the grabber in dconf-editor going to org->gnome->feedreader and switch the setting "content-grabber" to false. Then set up your account again.
If it works afterwards, can you post the feed-url of the website you are using (http://www.fel.cvut.czBylo) so I can figure out what is happening there?

**Comment by okias** [comment]
<outline title="Aktuality - FEL" text="Aktuality - FEL" type="rss" xmlUrl="http://www.fel.cvut.cz/aktuality/rss.xml" htmlUrl="http://www.fel.cvut.cz/aktuality/rss.xml"/>

**Comment by jangernert** [comment]
Jup, that should not happen. I will see if I can reproduce and fix the issue ;)

**Comment by okias** [comment]
ok, then just tell me, when I can remove & re-add "clear" FEL rss (I don't need history from this RSS channel)..

**Comment by jangernert** [comment]
@okias Small problem over here. Tiny Tiny RSS handles the feed without problems.
```
[ DEBUG ] Grabber: process article: http://www.fel.cvut.cz/cz/aktuality/promoce-srpen.html
[ DEBUG ] Grabber: no config (fel.cvut.cz.txt) found for article: http://www.fel.cvut.cz/cz/aktuality/promoce-srpen.html
[ DEBUG ] Grabber: no config (.cvut.cz.txt) - cutSubdomain - found for article: http://www.fel.cvut.cz/cz/aktuality/promoce-srpen.html
[ DEBUG ] GrabberUtils: repairURL xpath:"//img" attr:"src"
[ DEBUG ] GrabberUtils: save Images: 149418, 134
[ DEBUG ] GrabberUtils: postProcessing
[ DEBUG ] GrabberUtils: postProcessing done
```

Can you maybe add some sort of guest account to your owncloud-news setup, add the faulty feed and tell me how to access it? (jangernert@gmail.com).

**Comment by jangernert** [comment]
My local setup of [ocdev](https://github.com/owncloud/ocdev) works as well.
I added the feed manually with this link: http://www.fel.cvut.cz/aktuality/rss.xml


```
[ DEBUG ] Grabber: process article: http://www.fel.cvut.cz/cz/aktuality/cena-edwards
[ DEBUG ] Grabber: no config (fel.cvut.cz.txt) found for article: http://www.fel.cvut.cz/cz/aktuality/cena-edwards
[ DEBUG ] Grabber: no config (.cvut.cz.txt) - cutSubdomain - found for article: http://www.fel.cvut.cz/cz/aktuality/cena-edwards
[ DEBUG ] GrabberUtils: repairURL xpath:"//img" attr:"src"
[ DEBUG ] GrabberUtils: save Images: 117, 4
[ DEBUG ] GrabberUtils: postProcessing
[ DEBUG ] GrabberUtils: postProcessing done
[ DEBUG ] Grabber: process article: http://www.fel.cvut.cz/cz/aktuality/studenti-otevrene-informatiky-ziskali-oceni-upe
[ DEBUG ] Grabber: no config (fel.cvut.cz.txt) found for article: http://www.fel.cvut.cz/cz/aktuality/studenti-otevrene-informatiky-ziskali-oceni-upe
[ DEBUG ] Grabber: no config (.cvut.cz.txt) - cutSubdomain - found for article: http://www.fel.cvut.cz/cz/aktuality/studenti-otevrene-informatiky-ziskali-oceni-upe
[ DEBUG ] GrabberUtils: repairURL xpath:"//img" attr:"src"
[ DEBUG ] GrabberUtils: save Images: 118, 4
[ DEBUG ] GrabberUtils: postProcessing
[ DEBUG ] GrabberUtils: postProcessing done
[ DEBUG ] Utils: generate preview for article: 111
[ DEBUG ] Utils: generate preview for article: 112
[ DEBUG ] Utils: generate preview for article: 113
[ DEBUG ] Utils: generate preview for article: 114
[ DEBUG ] Utils: generate preview for article: 115
[ DEBUG ] Utils: generate preview for article: 116
[ DEBUG ] Utils: generate preview for article: 117
[ DEBUG ] Utils: generate preview for article: 118
[ DEBUG ] FeedServer: new articles: 8
[ INFO  ] daemon: sync finished
[ DEBUG ] OwnCloud URL: http://localhost:8080/index.php/apps/news/api/v1-2/
```

**Comment by okias** [comment]
you should have access to my nextcloud instance.. new RSS works, it's old article which won't get loaded....

**Comment by ZanderBrown** [comment]
in a side note why are we `org.gnome.feedreader` because last i checked we are not part of gnome. Surely it should be `io.github.jangernert.feedreader`?

**Comment by jangernert** [comment]
@okias This character in the article-URL was causing the crash in libsoup
> ř

This commit should fix the issue b0e11bf625d1f39076edef6385d4ae22287ddfda.

@ZanderBrown tbh, when I started writing FeedReader I had no idea what the pattern of the gsettings-keys was and I just copied them from other gnome applications.

**Comment by ZanderBrown** [comment]
@jangernert i though that might be it. It's not ideal but it could be worse
